oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureDataStore.java

```diff
@@ -41,17 +42,24 @@
 
     protected Properties properties;
 
+    private SasTokenGenerator sasTokenGenerator;
+
     private AzureBlobStoreBackend azureBlobStoreBackend;
 
     @Override
     protected AbstractSharedBackend createBackend() {
         azureBlobStoreBackend = new AzureBlobStoreBackend();
         if (null != properties) {
             azureBlobStoreBackend.setProperties(properties);
+            azureBlobStoreBackend.setSasTokenGenerator(sasTokenGenerator);
```

Comment:
This doesn't look right to me. Why set the `sasTokenGenerator` only when there are `properties`?

Reply:
Right. It should be outside the `if` clause.